zone/bonuses.cpp

```diff
@@ -184,405 +219,554 @@ void Client::CalcItemBonuses(StatBonuses* newbon) {
+				if (item->BardValue > b->brassMod) {
+					b->brassMod = item->BardValue;
+				}
+
+				break;
+			}
+			case EQ::item::ItemTypePercussionInstrument: {
+				if (item->BardValue > b->percussionMod) {
+					b->percussionMod = item->BardValue;
+				}
+
+				break;
+			}
+		}
+
+		if (item->SkillModValue != 0 && item->SkillModType <= EQ::skills::HIGHEST_SKILL) {
+			if (
+				(item->SkillModValue > 0 && b->skillmod[item->SkillModType] < item->SkillModValue) ||
+				(item->SkillModValue < 0 && b->skillmod[item->SkillModType] > item->SkillModValue)
+				) {
+				b->skillmod[item->SkillModType] = item->SkillModValue;
+			}
 		}
 
-		// Add Item Faction Mods
 		if (item->FactionMod1) {
 			if (item->FactionAmt1 > 0 && item->FactionAmt1 > GetItemFactionBonus(item->FactionMod1)) {
 				AddItemFactionBonus(item->FactionMod1, item->FactionAmt1);
-			}
-			else if (item->FactionAmt1 < 0 && item->FactionAmt1 < GetItemFactionBonus(item->FactionMod1)) {
+			} else if (item->FactionAmt1 < 0 && item->FactionAmt1 < GetItemFactionBonus(item->FactionMod1)) {
 				AddItemFactionBonus(item->FactionMod1, item->FactionAmt1);
 			}
 		}
+
 		if (item->FactionMod2) {
 			if (item->FactionAmt2 > 0 && item->FactionAmt2 > GetItemFactionBonus(item->FactionMod2)) {
 				AddItemFactionBonus(item->FactionMod2, item->FactionAmt2);
-			}
-			else if (item->FactionAmt2 < 0 && item->FactionAmt2 < GetItemFactionBonus(item->FactionMod2)) {
+			} else if (item->FactionAmt2 < 0 && item->FactionAmt2 < GetItemFactionBonus(item->FactionMod2)) {
 				AddItemFactionBonus(item->FactionMod2, item->FactionAmt2);
 			}
 		}
+
 		if (item->FactionMod3) {
 			if (item->FactionAmt3 > 0 && item->FactionAmt3 > GetItemFactionBonus(item->FactionMod3)) {
 				AddItemFactionBonus(item->FactionMod3, item->FactionAmt3);
-			}
-			else if (item->FactionAmt3 < 0 && item->FactionAmt3 < GetItemFactionBonus(item->FactionMod3)) {
+			} else if (item->FactionAmt3 < 0 && item->FactionAmt3 < GetItemFactionBonus(item->FactionMod3)) {
 				AddItemFactionBonus(item->FactionMod3, item->FactionAmt3);
 			}
 		}
+
 		if (item->FactionMod4) {
 			if (item->FactionAmt4 > 0 && item->FactionAmt4 > GetItemFactionBonus(item->FactionMod4)) {
 				AddItemFactionBonus(item->FactionMod4, item->FactionAmt4);
-			}
-			else if (item->FactionAmt4 < 0 && item->FactionAmt4 < GetItemFactionBonus(item->FactionMod4)) {
+			} else if (item->FactionAmt4 < 0 && item->FactionAmt4 < GetItemFactionBonus(item->FactionMod4)) {
 				AddItemFactionBonus(item->FactionMod4, item->FactionAmt4);
 			}
 		}
 
 		if (item->ExtraDmgAmt != 0 && item->ExtraDmgSkill <= EQ::skills::HIGHEST_SKILL) {
-			if (
-				RuleI(Character, ItemExtraDmgCap) >= 0 &&
-				(newbon->SkillDamageAmount[item->ExtraDmgSkill] + item->ExtraDmgAmt) > RuleI(Character, ItemExtraDmgCap)
-			) {
-				newbon->SkillDamageAmount[item->ExtraDmgSkill] = RuleI(Character, ItemExtraDmgCap);
+			if (item->ExtraDmgSkill == ALL_SKILLS) {
+				const EQ::skills::SkillType a[9] = {
```

Comment:
Should this just be returned in a getter somewhere describing the list instead of inlined

eg GetExtraDmgSkillTypes